app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  after_create :update_user_posts_counter
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_user_posts_counter
+    author.update_column(:posts_counter, author.posts_counter.to_i + 1)
+  end
```

Comment:
- [ ] **[OPTIONAL]** You can use `increment!` to auto-update the `posts_counter`

  ```
  def update_user_posts_counter
    author.increment!(:posts_counter)
  end
  ```